sources/balanced_dollar.move

```diff
@@ -1,179 +1,202 @@
-        //     rollback,
-        //     sources,
-        //     destinations
-        // );
-        sendCallMessage(coin, vars.iconBnUSD, xcallMessage, rollback, sources, destinations, ctx );
+        let xcallMessage = cross_transfer::encode(&xcallMessageStruct, CROSS_TRANSFER);
+        let rollback = cross_transfer_revert::encode(&rollbackStruct, CROSS_TRANSFER_REVERT);
+        
+        let envelope = envelope::wrap_call_message_rollback(xcallMessage, rollback, sources, destinations);
+        xcall::send_call(xcall_state, fee, idcap, config.icon_bnusd, envelope::encode(&envelope), ctx);
     }
 
-    //todo: remove this method once xcall integrated
-    public fun sendCallMessage(coin: Coin<SUI>,  iconBnUSD: String, xcallMessage: XCrossTransfer, rollback: XCrossTransferRevert, sources: vector<String>, destinations: vector<String>, ctx: &mut TxContext){
-        debug::print(&iconBnUSD);
-        debug::print(&xcallMessage);
-        debug::print(&rollback);
-        debug::print(&sources);
-        debug::print(&destinations);
+    entry fun execute_call<BALANCED_DOLLAR>(carier: &mut TreasuryCapCarrier<BALANCED_DOLLAR>, config: &Config, xcall_manager_config: &XcallManagerConfig, xcall:&mut XCallState, fee: Coin<SUI>, request_id:u128, data:vector<u8>, ctx:&mut TxContext){
+        let idcap = xcall_manager::get_idcap(xcall_manager_config);
+        let ticket = xcall::execute_call(xcall, idcap, request_id, data, ctx);
+        let msg = execute_ticket::message(&ticket);
+        let from = execute_ticket::from(&ticket);
+        let protocols = execute_ticket::protocols(&ticket);
 
-        transfer::public_transfer(coin, tx_context::sender(ctx));
-    }
-
-    public fun handleCallMessage(
-        vars: &BalancedDollarVars,
-        xcallManagerVars: &XcallManagerVars,
-        from: String,
-        data: vector<u8>,
-        protocols: vector<String>
-    ) {
-        let verified = xcall_manager::verifyProtocols(xcallManagerVars, protocols);
+        let verified = xcall_manager::verify_protocols(xcall_manager_config, &protocols);
         assert!(
             verified,
             ProtocolMismatch
         );
 
-        //string memory method = data.getMethod();
-        let method = CROSS_TRANSFER;
+        let method: vector<u8> = cross_transfer::get_method(&msg);
+        assert!(
+            method == CROSS_TRANSFER || method == CROSS_TRANSFER_REVERT, 
+            UnknownMessageType
+        );
 
-        assert!(method == CROSS_TRANSFER || method == CROSS_TRANSFER_REVERT, UnknownMessageType);
         if (method == CROSS_TRANSFER) {
-            assert!(from == vars.iconBnUSD, OnlyICONBnUSD);
-            // Messages.XCrossTransfer memory message = data.decodeCrossTransfer(); 
-            // (,string memory to) = message.to.parseNetworkAddress();
-            // _mint(to.parseAddress("Invalid account"), message.value);
+            assert!(from == network_address::from_string(config.icon_bnusd), OnlyICONBnUSD);
```

Comment:
we might not need BALANCED_DOLLAR type in execute_call because treasuryCap is already concrete.